### ISSUE TITLE
the program closes at startup

### ISSUE THREAD

**Issue by Jeanoo** [stefan-niedermann/nextcloud-deck#705]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

**Describe the bug**

**To reproduce**
Issue is reproducible with an account from [try.nextcloud.com](https://try.nextcloud.com):
- [ ] Yes
- [X ] No

1. Go to 'start-up'
4. it closes

**Versions**
 - Nextcloud: 19.0.2
 - Nextcloud Deck: 1.1.2
 - Nextcloud Android Deck: 1.9.7

**Smartphone (please complete the following information):**
- Device: 
- Android-Version: 
- App-Store:
  - [ ] Google Play Store
  - [ ] Google Play Store (Beta channel)
  - [X] F-Droid
  - [ ] Huawei AppGallery

**Are you using LDAP?**
- [ ] Yes
- [X] No

**Stacktrace**
```
Full Crash:

App Version: 1.9.7
App Version Code: 1009007
App Flavor: fdroid

Files App Version Code: 30130190

---

OS Version: 4.9.31(136)
OS API Level: 25
Device: OnePlus3T
Manufacturer: OnePlus
Model (and Product): ONEPLUS A3010 (OnePlus3)

---

java.lang.OutOfMemoryError: Could not allocate JNI Env
	at java.lang.Thread.nativeCreate(Native Method)
	at java.lang.Thread.start(Thread.java:730)
	at java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:941)
	at java.util.concurrent.ThreadPoolExecutor.processWorkerExit(ThreadPoolExecutor.java:1009)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1151)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:761)

```

**Comment by desperateCoder** [comment]
Hi @Jeanoo, thanks for your report!

I could need some more details: does this really happen every time you start the app? Has it ever worked for you?

I have the same device as you, but I got a custom ROM on it. Are you using OxygenOS?

**Comment by Jeanoo** [comment]
it happens every time I create a list followed by a card without an internet connection 
I do not use OxygenOS

**Comment by stefan-niedermann** [comment]
@Jeanoo can you please verify that it is each time the exact same error message when it crashes after your described steps?

```
java.lang.OutOfMemoryError
```

Or is it different (please post the other message in this case)?

**Comment by Jeanoo** [comment]
Hello,
I don't quite understand what you want to tell me.

Today, I've re-installed nextcloud and deck to get back on track.

without doing anything just update the page I have a series of problems, look: 

`Full Crash:

App Version: 1.11.1

App Version Code: 1011001

App Flavor: fdroid

Files App Version Code: 30130190

---

OS Version: 3.18.14-15260192(T580XXS5CTD1)

OS API Level: 27

Device: gtaxlwifi

Manufacturer: samsung

Model (and Product): SM-T580 (gtaxlwifixx)

---

android.view.WindowManager$BadTokenException: Unable to add window -- window android.view.ViewRootImpl$W@56df43 has already been added

at android.view.ViewRootImpl.setView(ViewRootImpl.java:1014)

at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:384)

at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:101)

at android.app.Dialog.show(Dialog.java:430)

at androidx.fragment.app.DialogFragment.onStart(DialogFragment.java:529)

at androidx.fragment.app.Fragment.performStart(Fragment.java:2730)

at androidx.fragment.app.FragmentStateManager.start(FragmentStateManager.java:365)

at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1194)

at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)

at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)

at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)

at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:447)

at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2169)

at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1992)

at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1947)

at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)

at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)

at android.os.Handler.handleCallback(Handler.java:790)

at android.os.Handler.dispatchMessage(Handler.java:99)

at android.os.Looper.loop(Looper.java:164)

at android.app.ActivityThread.main(ActivityThread.java:7000)

at java.lang.reflect.Method.invoke(Native Method)

at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:441)

at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1408)


```

```

Full Crash:

App Version: 1.11.1

App Version Code: 1011001

App Flavor: fdroid

Files App Version Code: 30130190

---

OS Version: 3.18.14-15260192(T580XXS5CTD1)

OS API Level: 27

Device: gtaxlwifi

Manufacturer: samsung

Model (and Product): SM-T580 (gtaxlwifixx)

---

android.database.CursorWindowAllocationException: Cursor window allocation of 2048 kb failed. # Open Cursors=1 (# cursors opened by this proc=1)

at android.database.CursorWindow.<init>(CursorWindow.java:108)

at android.database.AbstractWindowedCursor.clearOrCreateWindow(AbstractWindowedCursor.java:198)

at android.database.sqlite.SQLiteCursor.clearOrCreateWindow(SQLiteCursor.java:309)

at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:147)

at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:141)

at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:220)

at android.database.AbstractCursor.moveToNext(AbstractCursor.java:269)

at androidx.room.InvalidationTracker$1.checkUpdatedTable(InvalidationTracker.java:417)

at androidx.room.InvalidationTracker$1.run(InvalidationTracker.java:388)

at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)

at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)

at java.lang.Thread.run(Thread.java:764)


```

Full Crash:

App Version: 1.11.1

App Version Code: 1011001

App Flavor: fdroid

Files App Version Code: 30130190

---

OS Version: 3.18.14-15260192(T580XXS5CTD1)

OS API Level: 27

Device: gtaxlwifi

Manufacturer: samsung

Model (and Product): SM-T580 (gtaxlwifixx)

---

android.database.CursorWindowAllocationException: Cursor window allocation of 2048 kb failed. # Open Cursors=2 (# cursors opened by this proc=2)

at android.database.CursorWindow.<init>(CursorWindow.java:108)

at android.database.AbstractWindowedCursor.clearOrCreateWindow(AbstractWindowedCursor.java:198)

at android.database.sqlite.SQLiteCursor.clearOrCreateWindow(SQLiteCursor.java:309)

at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:147)

at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:141)

at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.dao.JoinCardWithUserDao*Impl.filterDeleted(JoinCardWithUserDao*Impl.java:451)

at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.DataBaseAdapter.filterRelationsForCard(DataBaseAdapter.java:142)

at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.DataBaseAdapter.filterRelationsForCard(DataBaseAdapter.java:177)

at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.DataBaseAdapter.lambda$7QItJ6JMTCZgTDMCP0GwmBwJWCQ(Unknown Source:0)

at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.-$$Lambda$DataBaseAdapter$7QItJ6JMTCZgTDMCP0GwmBwJWCQ.onDataChanged(Unknown Source:4)

at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.util.LiveDataHelper.lambda$null$0(LiveDataHelper.java:41)

at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.util.-$$Lambda$LiveDataHelper$mrHqtXbPmlW1ZeMzJCfhDExZ6hg.run(Unknown Source:6)

at java.lang.Thread.run(Thread.java:764)


````

**Comment by desperateCoder** [comment]
Oh boy, we're really sorry that you're facing this problems! 

Everything I can read out the error messages points to a memory issue, which is hard to cover, since we can't test the app on each end every hard- and software constellation out there. You already gave us some hints, but sorry if the analysis of this takes quite some time.

One more thing you could tell us: how much ram does your device have? And have you tried the app on a different device already?

**Comment by Jeanoo** [comment]
tell you where nextcoud (server) is installed

**Comment by Jeanoo** [comment]
I have just carried out the same operation
create a list followed by a map without internet connection, result Full Crash then duplicates

**Comment by stefan-niedermann** [comment]
> by a map without internet connection

I don't know what you mean with "a map without internet connection", can you please share a screenshot of this "map"?

---

> I don't quite understand what you want to tell me.

[I asked you](https://github.com/stefan-niedermann/nextcloud-deck/issues/705#issuecomment-719900474) if the error message always contains `OutOfMemoryError`. You sent [3 more samples](https://github.com/stefan-niedermann/nextcloud-deck/issues/705#issuecomment-720096415) which contain `CursorWindowAllocationException` and `WindowManager$BadTokenException`.

None of those error messages is a typical "programming" error (which we were able to localize and fix), they are more runtime related.
In other words: All of those error messages hint at a problem with your device itself (or device setup - for example a defect Custom ROM).

@desperateCoder therefore [wanted to know](https://github.com/stefan-niedermann/nextcloud-deck/issues/705#issuecomment-720097687) **whether you are able to reproduce any of those** multiple (but *different*) **errors on another device** (for example borrow one from your family or friends for testing purposes).

**Comment by Jeanoo** [comment]
> I don't know what you mean with "a map without internet connection", can you please share a screenshot of this "map"?

I will try to explain myself better.
In the presentation of de deck android in https://f-droid.org/en/packages/it.niedermann.nextcloud.deck/
it is noted:
"Works offline."
When I create an "offline" card it works very well!
However, when I create an offline list followed by an offline card it doesn't work, I get a Full Crash and duplicates.

**Comment by Jeanoo** [comment]
>     I don't quite understand what you want to tell me.
> 
> I asked you if the error message always contains OutOfMemoryError. You sent 3 more samples which contain CursorWindowAllocationException and WindowManager$BadTokenException.
> 
> None of those error messages is a typical "programming" error (which we were able to localize and fix), they are more runtime related.
> In other words: All of those error messages hint at a problem with your device itself (or device setup - for example a defect Custom ROM).
> 
> @desperateCoder therefore wanted to know whether you are able to reproduce any of those multiple (but different) errors on another device (for example borrow one from your family or friends for testing purposes).

It's probably related to the fact that I have a huge list and maps.
Lists: about 50
maps: about 300
material: i have a raspberrypi 4 8GB with nextcloudpi distribution
Device: Iphone and galaxy tab and pc windows
all appliances are working well.

If my raspberry wouldn't work well, I would have problems outside the android deck application and this is not the case, except with the online version of ios there is a bug that prevents me from creating cards for all the iphones I tested.   

Translated with www.DeepL.com/Translator (free version)

**Comment by desperateCoder** [comment]
Ok, so I think your problem is related to the limited RAM (2GB, right?) of your device, not the raspberry. 

I think reading the changed cards fills up all memory, then your device starts to panic and won't give our app enough memory for your big load of data.

I'll see, if and what I can do about it. Thanks so far!

**Comment by Jeanoo** [comment]
Ok thank you !
my galaxy device has 2GB of ram

**Comment by desperateCoder** [comment]
@Jeanoo could you check, if this still happens with the latest version? By "latest" I mean something above version 1.12.2 (you can look this up in the about page of the Android Deck app), there should be an update on F-Droid already. Also please make sure, you have the latest Server-Deck app, which should be at version 1.2.2.

We made a bunch of optimizations, maybe thats already enough to save your phones memory from catching fire. :sweat_smile:

**Comment by desperateCoder** [comment]
Ping @Jeanoo 

> @Jeanoo could you check, if this still happens with the latest version? By "latest" I mean something above version 1.12.2 (you can look this up in the about page of the Android Deck app), there should be an update on F-Droid already. Also please make sure, you have the latest Server-Deck app, which should be at version 1.2.2.

**Comment by Jeanoo** [comment]
I'm sorry, but I don't use the application anymore.

**Comment by stefan-niedermann** [comment]
That's a pitty, thank you for the feedback, though 🙂.

We'll close the issue as not reproducible then because no other user ever reported it and no developer is able to reproduce it on any emulator or physical device.

In case you decide to give the app another try, let us know if you encounter any issue again.

Regards and a good start in the new year! 🙂
